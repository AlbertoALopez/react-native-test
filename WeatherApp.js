/*
 *  Main weather app react component
 * 
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

export default class WeatherApp extends Component {
  constructor() {
    super();
    this.state = {
      zip: '',
    };
  }
  _handleTextChange = (event) => {
    console.log(event.nativeEvent.text);
    this.setState({
      zip: event.nativeEvent.text,
    });
  }
  render() {
    return (
      <View style={styles.container}>
        <TextInput
          style={ styles.input }
          onSubmitEditing={ this._handleTextChange }
        />
        <Text style={styles.welcome}>
          You inputted { this.state.zip }.
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.android.js
        </Text>
        <Text style={styles.instructions}>
          Double tap R on your keyboard to reload,{'\n'}
          Shake or press menu button for dev menu
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  input: {
    fontSize: 20,
    borderWidth: 2,
    height: 40,
  },
});
