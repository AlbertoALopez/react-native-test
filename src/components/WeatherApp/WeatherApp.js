/*
 *  Main weather app react containers
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  View,
} from 'react-native';
import Forecast from '../Forecast/Forecast';


const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#4D4D4D',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  input: {
    alignSelf: 'stretch',
    fontSize: 20,
    borderWidth: 2,
    height: 40,
  },
});

export default class WeatherApp extends Component {
  constructor() {
    super();
    this.state = {
      zip: '',
      forecast: {
        main: 'Clouds',
        description: 'Slightly cloudy',
        temp: 24.1,
      },
    };
  }
  handleTextChange = (event) => {
    console.log(event.nativeEvent.text);
    this.setState({
      zip: event.nativeEvent.text,
    });
  }
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          You inputted {this.state.zip}.
        </Text>
        <Forecast
          main={this.state.forecast.main}
          description={this.state.forecast.description}
          temp={this.state.forecast.temp}
        />
        <TextInput
          style={styles.input}
          returnKeyType="go"
          onSubmitEditing={this.handleTextChange}
        />
      </View>
    );
  }
}
