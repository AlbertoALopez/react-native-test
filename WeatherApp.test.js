/*
 * Tests for main WeatherApp component
 */

import 'react-native';
import React from 'react';
import WeatherApp from './WeatherApp.js';
import renderer from 'react-test-renderer';


it('WeatherApp renders correctly', () => {
    const tree = renderer.create(
        <WeatherApp />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
