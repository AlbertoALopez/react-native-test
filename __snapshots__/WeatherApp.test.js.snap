exports[`test WeatherApp renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <TextInput
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderWidth": 2,
        "fontSize": 20,
        "height": 40,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    You input 
    
    .
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    To get started, edit index.android.js
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    Double tap R on your keyboard to reload,
    
    
    Shake or press menu button for dev menu
  </Text>
</View>
`;
